ir/src/constraints/graph.rs

```diff
@@ -360,9 +360,7 @@ impl AlgebraicGraph {
                         variable_expr,
                     )
                 } else {
-                    Err(SemanticError::InvalidUsage(format!(
-                        "Identifier {ident} was declared as a {elem_type} which is not a supported type."
-                    )))
+                    Err(SemanticError::unsupported_identifer_type(ident, elem_type))
```

Comment:
comment for the future: we should revisit this. The problem isn't that the types are unsupported generally. It's that they're not allowed in the place where they're being used. However, this section will be changing significantly during the refactor #150, so I think it's best to merge it as it is.